programs/benchmark/Benchmark.cpp

```diff
@@ -235,6 +238,8 @@ class Benchmark : public Poco::Util::Application
 
     std::atomic<size_t> queries_executed{0};
 
+    std::vector<size_t> queries_per_connection;
```

Comment:
The `queries_per_connection` doesn't look threadsafe.  It's checked and incremented inside `execute`, but multiple threads could possibly access and modify it simultaneously leading to potential race conditions.

Reply:
Makes sense.
I tried using `std::vector<std::atomic<size_t>>` but I was not sure if this is correct way to initialise it:
```c++
queries_per_connection = std::vector<std::atomic<size_t>>(connections.size());
for (auto & counter : queries_per_connection)
       counter.store(0);
```
This is the only way that was not giving compiler errors but I feel this is wrong cause this tries to copy-default-initialize `std::atomic<size_t>`, which is not allowed because `std::atomic` is not copyable.

So I used a mutex instead to provide thread safe access for `queries_per_connection`.

Reply:
Yeah using a mutex should work and be more simpler than using std::atomic for initialization.